clang/test/Frontend/macro_defined_type.cpp

```diff
@@ -14,12 +16,26 @@ void Func() {
   auto NODEREF auto_i2 = i; // expected-warning{{'noderef' can only be used on an array or pointer type}}
 }
 
+// The diagnostic message is hard-coded as 'noderef' so using a system macro doesn't change the behavior
+void Func_system_macro() {
+  int _SYS_NODEREF i; // expected-warning{{'noderef' can only be used on an array or pointer type}}
+  int _SYS_NODEREF *i_ptr;
+
+  auto _SYS_NODEREF *auto_i_ptr2 = i_ptr;
+  auto _SYS_NODEREF auto_i2 = i; // expected-warning{{'noderef' can only be used on an array or pointer type}}
+}
+
+
 // Added test for fix for P41835
 #define _LIBCPP_FLOAT_ABI __attribute__((pcs("aapcs")))
 struct A {
   _LIBCPP_FLOAT_ABI int operator()() throw(); // expected-warning{{'pcs' calling convention is not supported for this target}}
 };
 
+struct A_system_macro {
+  _SYS_LIBCPP_FLOAT_ABI int operator()() throw(); // expected-warning{{'_SYS_LIBCPP_FLOAT_ABI' calling convention is not supported for this target}}
```

Comment:
This seems to be losing important information -- it's no longer possible to tell from this diagnostic which calling convention was being applied / is not supported.

Can we add an "aka" for these cases, eg: `'_SYS_LIBCPP_FLOAT_ABI' (aka 'pcs') calling convention is not supported for this target`

Reply:
@zygoloid Thanks for the comment! Right, the compiler actually does emit "note" like below, which I forgot to add in the test. I will add it. Would it address your concern?

```
clang/test/Frontend/Inputs/macro_defined_type.h:4:46: note: expanded from macro '_SYS_LIBCPP_FLOAT_ABI'
    4 | #define _SYS_LIBCPP_FLOAT_ABI __attribute__((pcs("aapcs")))
      |                                                                                       ^    
```

Reply:
Yes, if we're always pointing the caret at the attribute itself so we get an 'expanded from macro' note, that seems fine.

Reply:
Fwiw, my test may not be a good example. Since this is only for system-defined macro, so I'd expect the macro name should more helpful in real use cases.

Reply:
> if we're always pointing the caret at the attribute itself so we get an 'expanded from macro' note, that seems fine.

@zygoloid I just added some file checks here (411e3435873bad900e84309d90fe13f6085b3a1a) to verify that.